### PR TITLE
move duplicated binding generator visitor to it's own class

### LINKED ISSUE #1082: Pull binding code up into an `AbstractModule`
see: https://github.com/kframework/k/pull/1070#discussion-diff-19056637R103

### PR DESCRIPTION
**Pull request by osa1**
Resolves #1082.

### CODE REVIEW

**Comment by osa1** [comment]
@dwightguth @cos this is ready for reviews.

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/JavaSymbolicKRunModule.java [hunk 1]
```diff
@@ -92,7 +87,7 @@ Definition javaDefinition(BinaryLoader loader, Context context, FileUtil files,
         }
     }
 
-    public static class SimulationModule extends PrivateModule {
+    public static class SimulationModule extends PrivateKModule {
```
Comment (cos) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/JavaSymbolicKRunModule.java]: why is it called `PrivateKModule`?

Comment (osa1) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/JavaSymbolicKRunModule.java]: It has a minor addition to `PrivateModule` so I used a name that resembles it. What do you want me to call it?

Comment (dwightguth) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/JavaSymbolicKRunModule.java]: I'd recommend something like AnnotatedByDefinitionModule

**Comment by dwightguth** [comment]
lgtm other than the name

**Comment by osa1** [comment]
Renamed the class.

**Comment by cos** [comment]
lgtm

**Pull request merged**